docs/core/entity.md

```diff
@@ -376,6 +376,85 @@ Called when an entity has their entity_id and hass object assigned, before it is
 
 Called when an entity is about to be removed from Home Assistant. Example use: disconnect from the server or unsubscribe from updates.
 
+## Icons
+
+Every entity in Home Assistant has an icon, which is used as a visual indicator to identify the entity more easily in the frontend. Home Assistant uses the [Material Design Icons](https://materialdesignicons.com/) icon set.
+
+In most cases, Home Assistant will pick an icon automatically based on the entity's domain, `device_class`, and `state`. It is preferred to use the default icon if possible, to provide a consistent experience and to avoid confusion for the user. However, it is possible to override provide a custom icon for an entity.
+
+Regardless of the provided icon, it is always possible for the user to customize the icon to their liking in the frontend.
+
+There are two ways to provide a custom icon for an entity, either by providing icon translations or by providing an icon identifier.
+
+### Icon translations
+
+TThis is the preferred and most modern way to provide a custom icon for an entity. Icon translations work similarly [our regular translations](/docs/internationalization/core#state-of-entities), but instead of translating the state of an entity, they translate the states of an entity to icons.
+
+The `translation_key` property of an entity defines the icon translation to use. This property is used to look up the translation in the `entity` section of the integration's `icons.json` file.
+
+To differentiate entities and their translations, provide different translation keys. The following example `icons` for a Moon domain `sensor` entity with its `translation_key` property set to phase:
+
+```json
+{
+  "entity": {
+    "sensor": {
+      "phase": {
+        "default": "mdi:moon",
+        "state": {
+          "new_moon": "mdi:moon-new",
+          "first_quarter": "mdi:moon-first-quarter",
+          "full_moon": "mdi:moon-full",
+          "last_quarter": "mdi:moon-last-quarter"
+        }
+      }
+    }
+  }
+}
+```
+
+Notice that icons start with `mdi:` plus an [identifier](https://materialdesignicons.com/). The `default` icon is used when the entity's state is not in the `state` section. The `state` section is optional, and if not provided, the `default` icon will be used for all states.
+
+Icons for entity state attributes can also be provided. The following example icons for a thermostat providing a `climate` entity with its `translation_key` property set to `ubercool`. This entity has a `preset_mode` state attribute, which can be set to `vacation` or `night`. The frontend will use these in, for example, the climate card.
```

Comment:
"The following example icons for a thermostat providing a `climate` entity wi" doesn't sounds correct

Reply:
```suggestion
Icons for entity state attributes can also be provided. The following example provides icons for a `climate` entity with its `translation_key` property set to `ubercool`. This entity has a `preset_mode` state attribute, which can be set to `vacation` or `night`. The frontend will use these in, for example, the climate card.
```